src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs

```diff
@@ -1579,7 +1579,7 @@ private void AddUpdate(ZipUpdate update)
 		{
 			contentsEdited_ = true;
 
-			int index = FindExistingUpdate(update.Entry.Name);
+			int index = FindExistingUpdate(update.Entry);
```

Comment:
Why is FindExistingUpdate changed here?

The `FindExistingUpdate(ZipEntry)` method should probably just call `FindExistingUpdate(string)` for DRYness, but that's for another issue...